src/java.base/share/classes/jdk/internal/foreign/NativeMemorySegmentImpl.java

```diff
@@ -36,7 +36,7 @@
  * Implementation for native memory segments. A native memory segment is essentially a wrapper around
  * a native long address.
  */
-sealed class NativeMemorySegmentImpl extends AbstractMemorySegmentImpl permits MappedMemorySegmentImpl {
+public sealed class NativeMemorySegmentImpl extends AbstractMemorySegmentImpl permits MappedMemorySegmentImpl {
```

Comment:
Why does this class have to be `public`? Maybe it was a change that survived from the larger version of the PR? The same goes for `MappedMemorySegmentImpl`.